### PR TITLE
Docs, units, autogen.

### PR DESCRIPTION
**Pull request by sofar**
4 changes, all fairly trivial.

1. Add auto-update units. Reworks configure.ac to not create non-dynamic files.
2. Add trivial autogen.sh
3. Add man page support.
4. Add (stored) generated troff files.

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -111,16 +113,9 @@ systemdunitdir = @SYSTEMD_UNITDIR@
...
-systemdunit-install-local:
-	mkdir -p $(DESTDIR)$(systemdunitdir)/multi-user.target.wants/
-	ln -sf ../check-update.timer $(DESTDIR)$(systemdunitdir)/multi-user.target.wants/check-update.timer
-install-data-local: systemdunit-install-local
-
-systemdunit-uninstall-local:
-	rm -f $(DESTDIR)$(systemdunitdir)/multi-user.target.wants/check-update.timer
-uninstall-local: systemdunit-uninstall-local
+        data/check-update.timer \
+	data/swupd-update.service \
```
Comment (bryteise) [inline Makefile.am]: Looks like whitespace damage here.

File: Makefile.am [hunk 2]
```diff
@@ -216,3 +216,21 @@ release:
...
+	docs/swupd.1 \
+	docs/check-update.service.4 \
+	docs/check-update.timer.4 \
+	docs/swupd-update.service.4 \
+	docs/swupd-update.timer.4 \
+	docs/update-triggers.target.4
+
+manpages:
+	for MANPAGE in $(MANPAGES); do \
+		ronn --roff < $${MANPAGE}.md > $${MANPAGE}; \
```
Comment (bryteise) [inline Makefile.am]: Maybe add a configure check for ronn?

File: Makefile.am (hunk 1 above)
Comment (sofar) [inline Makefile.am]: Aye, fixing.

File: Makefile.am (hunk 2 above)
Comment (sofar) [inline Makefile.am]: no part of the actual build will reach this target... I don't see a need to?

**Comment by sofar** [comment]
There was already some whitespace damage in `Makefile.am` I think. Fixed now.

File: Makefile.am (hunk 2 above)
Comment (bryteise) [inline Makefile.am]: Yea that's fine. Just was thinking of a warning for folks that building docs required it.

**Comment by bryteise** [comment]
+1 from me

**Pull request merged**